Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -89,6 +89,59 @@ namespace AZ::Dom
         return !operator==(key);
     }
 
+    bool PathEntry::operator<(const PathEntry& other) const
+    {
+        const size_t* otherIndexValue = AZStd::get_if<size_t>(&other.m_value);
+        const AZ::Name* otherNameValue = AZStd::get_if<AZ::Name>(&other.m_value);
+        if (otherIndexValue)
+        {
+            return operator<(*otherIndexValue);
+        }
+        return operator<(*otherNameValue);
```

Comment:
Is there a better way to do this comparison?

Reply:
Actually here is what can be done.
If the `m_value` variant contains a size_t, convert it to a string.
And then just use the `string_view` < operator

```suggestion
        // Used to store a string that has been from an index path entry
        // for long enough to be used in a string comparison 
        AZStd::fixed_string<32> thisIndexStorage;
        AZStd::fixed_string<32> otherIndexStorage;
        
        // References a view of this path entry as a string
        AZStd::string_view thisValueView;
        // References a view of the other path entry as a string
        AZStd::string_view otherValueView;
        
        // Get string_view for this PathEntry instance
        if (const size_t* thisIndexValue = AZStd::get_if<size_t>(&m_value);
            thisIndexValue != nullptr)
        {
            AZStd::to_string(thisIndexStorage, *thisIndexValue);
            thisValueView = thisIndexStorage;
        }
        else if (const AZ::Name* thisNameValue = AZStd::get_if<AZ::Name>(&m_value);
            thisNameValue != nullptr)
        {
            thisValueView = thisNameValue->GetStringView();
        }

        if (const size_t* otherIndexValue = AZStd::get_if<size_t>(&other.m_value);
            otherIndexValue != nullptr)
        {
            AZStd::to_string(otherIndexStorage, *otherIndexValue);
            otherValueView = otherIndexStorage;
        }
        else if (const AZ::Name* otherNameValue = AZStd::get_if<AZ::Name>(&other.m_value);
            otherNameValue != nullptr)
        {
            otherValueView = otherNameValue->GetStringView();
        }

        return thisValueView < otherValueView;
```

Reply:
This is cool and clever, but it looks like I might need the other operator< overloads anyway for value comparisons like:
Path("0/1").GetEntries()[1] < "donut"
Path("0/3/1").GetEntries()[1] < "2"

I'll leave it as is for now since I do need those other operators anyway.